### ISSUE TITLE
Bug: Meal Plan does not render `Add all to shopping list` in UK timezone

### ISSUE THREAD

**Issue by userbradley** [grocy/grocy#2168]
### Previous Interactions
- [Reddit post](https://www.reddit.com/r/grocy/comments/106skts/no_put_mising_products_on_shopping_list_since/)

### Possible linked issues
- https://github.com/grocy/grocy/issues/214
- https://github.com/grocy/grocy/issues/808

## Describe the issue

Since the year rolled over in the UK (2022 -> 2023) on my meal planner the `Add missing items to shopping list` for the week has disappeared 

## Show me the issue

When putting a breakpoint on line `77` of `mealplan.js` we are able to get the values 

![image](https://user-images.githubusercontent.com/41597815/225598787-4e1200a5-277c-4789-9dc5-6a0d6a78cd2b.png)

The issue here is that js claims it's `2023-09` where as PHP claims it's `2023-10`

When I manually change the `weekRecipieName` value to be `2023-10` it renders the `Week cost` section 

![image](https://user-images.githubusercontent.com/41597815/225599044-aa3ec3a5-1622-4428-b603-0f6de34175c1.png)

### Are you the only person having this issue

No:

 @adamxp12 - In the UK Timezone and has it installed **not** using docker

### What other testing have you done

I have manually changed the below file and removed the  `-1` and it worked

```js
var weekRecipeName = view.start.year().toString() + "-" + ((view.start.week()).toString().padStart(2, "0")).toString();
```

https://github.com/grocy/grocy/blob/3f2ad17460d7b3b06f4c7c25243d1f9ee238fd80/public/viewjs/mealplan.js#L76

with my change, years+ from current works, until 2026
![image](https://user-images.githubusercontent.com/41597815/225599713-de8f606a-6c85-48d4-b708-d8743236b5d7.png)

## What troubleshooting steps have I tried

- Installing a whole new server on a different cloud platform
- Checking `api/system/time` all line up (they do)

![image](https://user-images.githubusercontent.com/41597815/225601148-382b3e7a-428c-449f-a9d8-4a8925f95fbc.png)

**Comment by berrnd** [comment]
And since it's obvously about week numbers, the most important info is the `CALENDAR_FIRST_DAY_OF_WEEK` and `MEAL_PLAN_FIRST_DAY_OF_WEEK` setting and if that's empty, the related settings based on your server and client locale.

**Comment by adamxp12** [comment]
Both of them values are set to 1 in my setup which is also broken in the exact same way. 1 being a Monday which is the UK first day of the week at least according to all the calendars I have seen

Find attached my PHP info() on date/time as I believe its the PHP date module thats at fault here. Looks to be set to UTC.

![image](https://user-images.githubusercontent.com/6089459/225606550-4d761e09-b537-487e-8468-6d12bbf02f38.png)

**Comment by berrnd** [comment]
> I believe its the PHP date module thats at fault here. Looks to be set to UTC.

Well, I don't live in the UK, but from what I've heard so far and as of today, UTC(+0) is the timezone used there. The requirement, mentioned on all places where time related problems were discussed so far, is that the server (= PHP and system settings) and the client (= Browser) needs to use the same timezone.

The screenshot of the API endpoint `/system/time` indiciates that that's the case for the PHP and system setting, this screenshot of `phpinfo()` indicates that thatÄs the case for the PHP setting. Haven't seen something client / Browser related here so far.

**Comment by wsetws** [comment]
Just to report the same issue on my end. My server is running since year and it used to have the "All to shopping list" button. I just noticed the button is not there days ago (but I was not actively using it since last year).

I am currently using the latests linuxserver.io docker container. (Grocy version: 3.3.2.  I updated from 3.3.1 which had the same issue).
Both `CALENDAR_FIRST_DAY_OF_WEEK`  and `MEAL_PLAN_FIRST_DAY_OF_WEEK` are set to 1.
My timezones looks the correct in the `/system/time` endpoint. I am at UTC-3.  (Docker host ,hypervisor and client timezone also are the same).  One extra detail I can think of is that we use daylight saving in my region. Not sure if that could impact somehow.
